File: java/PlanesAndroid/app/src/main/java/com/planes/android/GameStages.java
package com.planes.android;

public enum GameStages {
    GameNotStarted(2),
    BoardEditing(0),
    Game(1);

    private final int value;

    GameStages(final int newValue) {
        value = newValue;
    }

    public int getValue() {
        return value;
    }
}

File: kotlin/PlanesAndroid/app/src/main/java/com/planes/single_player_engine/GameStages.kt
package com.planes.single_player_engine

enum class GameStages(val value: Int) {
    GameNotStarted(0), BoardEditing(1), Game(2), WaitForOpponentPlanesPositions(3), WaitForOpponentMoves(4),
    SendRemainingMoves(5);

    companion object {
        private val map = values().associateBy(GameStages::value)
        operator fun get(value: Int) = map[value]
    }

}